### PR TITLE
@ #20 | should update README file

### PR DESCRIPTION
**Pull request by hieptranquoc**
connect #20

### CODE REVIEW

**Comment by hieptranquoc** [comment]
@hoatle review plz

File: README.md [hunk 1]
```diff
@@ -16,14 +10,79 @@ Kubernetes stack cookbook to work with Kubernetes
 
   ```bash
   $ cd ~/teracy-dev/workspace
-  $ git checkout <your_forked_repo>
+  $ git clone git@github.com:<your_forked_repo>/kubernetes-stack-cookbook.git
```
Comment (hoatle) [inline README.md]: git clone <your_forked_repo_address>

File: README.md [hunk 2]
```diff
@@ -16,14 +10,79 @@ Kubernetes stack cookbook to work with Kubernetes
...
   $ cd kubernetes-stack-cookbook
   $ git remote add upstream git@github.com:teracyhq-incubator/kubernetes-stack-cookbook.git
   ```
 
-- `$ vagrant reload --provision` to update the dev-setup from this project into the teracy-dev's VM.
+- remember to run `$ vagrant reload --provision` to update the dev-setup from this project into the teracy-dev's VM.
   After that, you should be ready to work on the project.
 
+## How to use
```
Comment (hoatle) [inline README.md]: no no, this is a standard Chef cookbook, follow README format from other Chef cookbooks, for example: https://github.com/chef-cookbooks/docker

**Review by hoatle — CHANGES_REQUESTED** [review]

File: README.md (hunk 2 above)
Comment (hieptranquoc) [reply README.md]: @hoatle standard Chef cookbook ? it's a standalone chef cookbook( run without teracy-dev)
 or run with teracy-dev

Comment (hoatle) [reply README.md]: The README does not need to mention about teracy-dev at all.

File: README.md [hunk 3]
```diff
@@ -16,14 +10,36 @@ Kubernetes stack cookbook to work with Kubernetes
 
   ```bash
   $ cd ~/teracy-dev/workspace
-  $ git checkout <your_forked_repo>
+  $ git clone git@github.com:<your_forked_repo_address>/kubernetes-stack-cookbook.git
```
Comment (hoatle) [inline README.md]: $ git checkout <your_forked_repo_address> is enough

File: dev-setup/vagrant_config_override.example.json [hunk 4]
```diff
@@ -0,0 +1,22 @@
+{
```
Comment (hoatle) [inline dev-setup/vagrant_config_override.example.json]: remove this file

File: dev-setup/vagrant_config_default.json [hunk 5]
```diff
@@ -9,6 +9,21 @@
       "kubernetes-stack-cookbook"
     ],
     "json": {
+      // to work with kubernetes
```
Comment (hoatle) [inline dev-setup/vagrant_config_default.json]: don't add this

File: dev-setup/chef/run-dev.sh [hunk 6]
```diff
@@ -2,13 +2,13 @@
...
 # install git first if not exist yet
 # see: https://stackoverflow.com/questions/592620/check-if-a-program-exists-from-a-bash-script
-# if hash git 2>/dev/null; then
-#     echo "git was installed";
-# else
-#     echo "installing git...";
-#     apt-get update;
-#     apt-get install -qy git
-# fi
+if hash git 2>/dev/null; then
```
Comment (hoatle) [inline dev-setup/chef/run-dev.sh]: no no

**Review by hoatle — CHANGES_REQUESTED** [review]
@hieptranquoc please update README with the following outline:

```
## Getting started

//TODO

## Platform support

//TODO

## Resources overview
//TODO

## How to develop
```

update README only, don't introduce new files or touch other files.

File: dev-setup/chef/run-dev.sh (hunk 6 above)
Comment (hieptranquoc) [reply dev-setup/chef/run-dev.sh]: fixed!

File: dev-setup/vagrant_config_default.json (hunk 5 above)
Comment (hieptranquoc) [reply dev-setup/vagrant_config_default.json]: fixed!

File: dev-setup/vagrant_config_override.example.json (hunk 4 above)
Comment (hieptranquoc) [reply dev-setup/vagrant_config_override.example.json]: fixed!

File: README.md (hunk 3 above)
Comment (hieptranquoc) [reply README.md]: fixed!

**Comment by hieptranquoc** [comment]
@hoatle check this PR again,plz

File: README.md [hunk 7]
```diff
@@ -2,13 +2,38 @@
...
+- Ubuntu(16.04)
+- Centos7 or higher
 
 ## How to use
 
-//TODO
+- Add `depends 'kubernetes-stack'` to your cookbook's metadata.rb
+- Use the resources shipped in cookbook in a recipe :
+
+```ruby
```
Comment (hoatle) [inline README.md]: @hieptranquoc what's about `binary_path` option for all there resources?

File: README.md [hunk 8]
```diff
@@ -2,13 +2,38 @@
 
 Kubernetes stack cookbook to work with Kubernetes
 
+## Requirements
+
+- Chef 12.5.x or higher. Chef 11 is NOT SUPPORTED.
+
+## Platform Support
```
Comment (hoatle) [inline README.md]: "Platform support", make sure everything is consistent.

File: README.md [hunk 9]
```diff
@@ -49,6 +71,12 @@ Kubernetes stack cookbook to work with Kubernetes
   $ kitchen verify <instance>
   ```
 
+## Resources Overview
+
+- [gcloud](#gcloud): install or remove google-cloud-sdk
+- [kubectl](#kubectl): install or remove kubectl
+- [helm](#helm): install or remove kubernetes/helm
```
Comment (hoatle) [inline README.md]: install or remove ``helm``

File: README.md [hunk 10]
```diff
@@ -49,6 +71,12 @@ Kubernetes stack cookbook to work with Kubernetes
   $ kitchen verify <instance>
   ```
 
+## Resources Overview
+
+- [gcloud](#gcloud): install or remove google-cloud-sdk
```
Comment (hoatle) [inline README.md]: install or remove ``google-cloud-sdk``

File: README.md [hunk 11]
```diff
@@ -49,6 +71,12 @@ Kubernetes stack cookbook to work with Kubernetes
...
+- [kubectl](#kubectl): install or remove kubectl
```
Comment (hoatle) [inline README.md]: ``kubectl``

File: README.md (hunk 10 above)
Comment (hoatle) [reply README.md]: highlight the programs with in ``

**Review by hoatle — CHANGES_REQUESTED** [review]
review done, @hieptranquoc please see my comments.

File: README.md (hunk 8 above)
Comment (hieptranquoc) [reply README.md]: @hoatle fixed!

File: README.md (hunk 7 above)
Comment (hieptranquoc) [reply README.md]: fixed!

File: README.md (hunk 10 above)
Comment (hieptranquoc) [reply README.md]: fixed!

File: README.md (hunk 11 above)
Comment (hieptranquoc) [reply README.md]: fixed!

File: README.md (hunk 9 above)
Comment (hieptranquoc) [reply README.md]: fixed!

File: README.md [hunk 12]
```diff
@@ -49,6 +80,12 @@ Kubernetes stack cookbook to work with Kubernetes
   $ kitchen verify <instance>
   ```
 
+## Resources Overview
```
Comment (hoatle) [inline README.md]: Resources overview

File: README.md [hunk 13]
```diff
@@ -2,13 +2,47 @@
 
 Kubernetes stack cookbook to work with Kubernetes
 
+## Requirements
 
-## How to use
+- Chef 12.5.x or higher. Chef 11 is NOT SUPPORTED.
+
+## Platform Support
```
Comment (hoatle) [inline README.md]: Platform support (the 2nd time and no changes after my 1st comment, this is bad)

Comment (hoatle) [reply README.md]: Using the lower, Upper case must be consistent with the same heading levels.

**Review by hoatle — CHANGES_REQUESTED** [review]

File: README.md [hunk 14]
```diff
@@ -2,13 +2,47 @@
...
+## How to use
 
+- Add `depends 'kubernetes-stack'` to your cookbook's metadata.rb
+- Use the resources shipped in cookbook in a recipe :
+
+```ruby
+kubectl 'install kubectl' do
+  action [:install, :remove]
+  version ''
+  binary_path '' #application location (if empty,default:/usr/local/bin/kubectl)
```
Comment (hoatle) [inline README.md]: if empty, default (1 white space after the comma character)

File: README.md [hunk 15]
```diff
@@ -2,13 +2,47 @@
...
+end
+
+gcloud 'install gcloud' do
+  action [:install, :remove]
+  version ''
+  binary_path '' #application location (if empty,default:/usr/local/bin/gcloud)
```
Comment (hoatle) [inline README.md]: if empty, default (1 white space after the comma character)

File: README.md [hunk 16]
```diff
@@ -2,13 +2,47 @@
...
+end
+
+helm 'install helm' do
+  action [:install, :remove]
+  version ''
+  binary_path '' #application location (if empty,default:/usr/local/bin/helm)
```
Comment (hoatle) [inline README.md]: if empty, default (1 white space after the comma character)

**Review by hoatle — CHANGES_REQUESTED** [review]
@hieptranquoc review done, pls check my comments.

File: README.md (hunk 16 above)
Comment (hieptranquoc) [reply README.md]: @hoatle fixed!

File: README.md (hunk 15 above)
Comment (hieptranquoc) [reply README.md]: fixed!

File: README.md (hunk 14 above)
Comment (hieptranquoc) [reply README.md]: fixed!

File: README.md (hunk 12 above)
Comment (hieptranquoc) [reply README.md]: fixed!

File: README.md (hunk 13 above)
Comment (hieptranquoc) [reply README.md]: fixed!

**Comment by hieptranquoc** [comment]
@hoatle fixed!

**Review by hoatle — APPROVED** [review]
LGTM

**Pull request merged**